### PR TITLE
A random assortment of things

### PR DESCRIPTION
**Pull request by SquidDev**
I'm sorry for the random assortment of changes: this was mostly me going through any unmapped methods/classes I was using and trying to find something for them.

I've tried to keep each commit as atomic as possible, so if you'd prefer, you can cherry-pick individual ones in. There's definitely a few here I'm less sure about, so suggestions are welcome.

### CODE REVIEW

File: mappings/net/minecraft/util/math/Polar2f.mapping [hunk 1]
```diff
@@ -1,5 +1,7 @@
 CLASS cjx net/minecraft/util/math/Polar2f
+	FIELD a FORWARDS Lcjx;
```
Comment (asiekierka) [inline mappings/net/minecraft/util/math/Polar2f.mapping]: > I'm really unsure about this name, so comments welcome!

File: mappings/net/minecraft/client/gui/hud/InGameHud.mapping [hunk 2]
```diff
@@ -17,8 +21,13 @@ CLASS cnr net/minecraft/client/gui/hud/InGameHud
 	FIELD w hudBossBar Lcnz;
 	METHOD a draw (F)V
 	METHOD a addChatMessage (Liy;Lja;)V
+	METHOD a setOverlayMessage (Lja;Z)V
+	METHOD a setPlayingOverlay (Ljava/lang/String;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/client/gui/hud/InGameHud.mapping]: Is this the best name?

File: mappings/net/minecraft/block/Block.mapping [hunk 3]
```diff
@@ -209,6 +209,7 @@ CLASS bfo net/minecraft/block/Block
 	METHOD d isSimpleFullBlock (Lboh;Lazz;Let;)Z
 	METHOD e isFullBoundsCubeForCulling (Lboh;)Z
 	METHOD e getMapColor (Lboh;Lazz;Let;)Lcde;
+	METHOD f getFluidState (Lboh;)Lccz;
```
Comment (asiekierka) [inline mappings/net/minecraft/block/Block.mapping:212]: I'm not too sure on it. I've been trying to think of a better name for a long time.

File: mappings/net/minecraft/block/Waterlogged.mapping [hunk 4]
```diff
@@ -0,0 +1 @@
+CLASS bkg net/minecraft/block/Waterlogged
```
Comment (jamierocks) [inline mappings/net/minecraft/block/Waterlogged.mapping]: `WaterloggedBlock`?

File: mappings/net/minecraft/client/gui/hud/ChatHud.mapping [hunk 5]
```diff
@@ -5,7 +5,9 @@ CLASS cob net/minecraft/client/gui/hud/ChatHud
 	METHOD a addMessage (Lja;)V
 	METHOD a addMessage (Lja;I)V
 		ARG 1 message
+	METHOD a addMessage (Lja;IIZ)V
 	METHOD a clear (Z)V
+	METHOD b removeChatMessage (I)V
```
Comment (jamierocks) [inline mappings/net/minecraft/client/gui/hud/ChatHud.mapping]: To be consistent with `addMessage`, `removeMessage`?

File: mappings/net/minecraft/block/Waterlogged.mapping (hunk 4 above)
Comment (asiekierka) [reply mappings/net/minecraft/block/Waterlogged.mapping]: This is an interface, *not* a class.

Comment (asiekierka) [reply mappings/net/minecraft/block/Waterlogged.mapping]: I say Waterloggable.

File: mappings/net/minecraft/client/gui/hud/InGameHud.mapping (hunk 2 above)
Comment (asiekierka) [reply mappings/net/minecraft/client/gui/hud/InGameHud.mapping]: Specifically, what does it refer to? Jukeboxes?

Comment (SquidDev) [reply mappings/net/minecraft/client/gui/hud/InGameHud.mapping]: Yep. It's the "now playing: 13" label you get above the hotbar.

Comment (asiekierka) [reply mappings/net/minecraft/client/gui/hud/InGameHud.mapping]: setPlayingOverlay sounds wrong to me... it's either too vague or insufficiently so.

File: mappings/net/minecraft/util/math/Polar2f.mapping (hunk 1 above)
Comment (kashike) [reply mappings/net/minecraft/util/math/Polar2f.mapping]: This class is a generic two-dimensional float vector - this field should be `ZERO`, instance fields should be `x` and `y`.

**Comment by liach** [comment]
i recall that playing overlay method's hotbar overlay has rainbow color while regular overlays from chat packets are white

**Comment by SquidDev** [comment]
@liach Yep! `setRecordPlayingOverlay` just calls `setOverlayMessage` with the second parameter set to true (which then will apply a tint).

**Review by Prospector — APPROVED** [review]

**Pull request merged**